language/move-prover/stackless-bytecode-generator/src/livevar_analysis.rs

```diff
@@ -83,9 +108,66 @@ impl LiveVarAnalysisProcessor {
             &code,
             &cfg,
         );
-        let mut annotations = analyzer.post_process(&cfg, &code, state_map);
-        let new_bytecode = analyzer.possibly_transform_code(&mut annotations, code);
-        (new_bytecode, annotations)
+        analyzer.post_process(&cfg, &code, state_map)
+    }
+
+    fn eliminate_unused_vars(
```

Comment:
I am unable to understand what this function is doing.  What does it have to do with removing redundant assignment?

Reply:
Well, you have a lot of unused variables after the dead code elimination, and those are simply eliminated here. Look at the baselines to see the effect.

Reply:
I am not sure which baseline you are referring to.  There are diffs in tons of files.  Could you point me to a specific test output file whose diff is particularly revealing?

Reply:
Look at the livevars baselines. These are the tests for the code you are commenting on. Also, isn't it clear that after dead code elimination there will be a lot of unused locals and it makes sense to eliminate them?

Reply:
When I was reading the implementation of eliminate_unused_vars last night, I couldn't understand it.  But today I do.